persistence/sql/persister_consent.go

```diff
@@ -264,103 +319,117 @@ func (p *Persister) ConfirmLoginSession(ctx context.Context, id string, authenti
 	return sqlcon.HandleError(
 		p.Connection(ctx).Update(&consent.LoginSession{
 			ID:              id,
+			NID:             p.NetworkID(ctx),
 			AuthenticatedAt: sqlxx.NullTime(authenticatedAt),
 			Subject:         subject,
 			Remember:        remember,
 		}))
 }
 
 func (p *Persister) CreateLoginSession(ctx context.Context, session *consent.LoginSession) error {
+	session.NID = p.NetworkID(ctx)
 	return sqlcon.HandleError(p.Connection(ctx).Create(session))
 }
 
 func (p *Persister) DeleteLoginSession(ctx context.Context, id string) error {
-	return sqlcon.HandleError(
-		p.Connection(ctx).Destroy(
-			&consent.LoginSession{ID: id},
-		))
+	count, err := p.Connection(ctx).RawQuery("DELETE FROM hydra_oauth2_authentication_session WHERE id=? AND nid = ?", id, p.NetworkID(ctx)).ExecWithCount()
+	if count == 0 {
+		return errorsx.WithStack(x.ErrNotFound)
```

Comment:
You could introduce a new method DestroyWithNetwork() which internally uses QueryWith... + Delete()

Reply:
Is there a way to count the affected rows with Delete?